pkg/provider/consulcatalog/config.go

```diff
@@ -42,6 +43,22 @@ func (p *Provider) buildConfiguration(ctx context.Context, items []itemData) *dy
 
 			provider.BuildTCPRouterConfiguration(ctxSvc, confFromLabel.TCP)
 
+			tcpOrUDP = true
+		}
+
+		if len(confFromLabel.UDP.Routers) > 0 || len(confFromLabel.UDP.Services) > 0 {
+			err := p.buildUDPServiceConfiguration(ctxSvc, item, confFromLabel.UDP)
+			if err != nil {
+				logger.Error(err)
+				continue
```

Comment:
Are we sure that we do not want to try building the other parts of the config (HTTP, if it exists) if the UDP part fails?

(and same question for the TCP case L41 of course).